cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -545,424 +550,487 @@ public Node getThisNode() {
+
+      /**
+       * Forward a non-query plan to "receiver" using "client".
+       * @param plan a non-query plan
+       * @param client
+       * @param receiver
+       * @param header to determine which DataGroupMember of "receiver" will process the request.
+       * @return a TSStatus indicating if the forwarding is successful.
+       */
+      TSStatus forwardPlan (PhysicalPlan plan, AsyncClient client, Node receiver, Node header){
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        DataOutputStream dataOutputStream = new DataOutputStream(byteArrayOutputStream);
+        try {
+          plan.serializeTo(dataOutputStream);
+          AtomicReference<TSStatus> status = new AtomicReference<>();
+          ExecutNonQueryReq req = new ExecutNonQueryReq();
+          req.setPlanBytes(byteArrayOutputStream.toByteArray());
+          if (header != null) {
+            req.setHeader(header);
+          }
+          synchronized (status) {
+            client.executeNonQueryPlan(req, new ForwardPlanHandler(status, plan, receiver));
+            status.wait(RaftServer.connectionTimeoutInMS);
+          }
+          return status.get() == null ? StatusUtils.TIME_OUT : status.get();
+        } catch (IOException | TException e) {
+          TSStatus status = StatusUtils.INTERNAL_ERROR.deepCopy();
+          status.setMessage(e.getMessage());
+          logger
+              .error("{}: encountered an error when forwarding {} to {}", name, plan, receiver, e);
+          return status;
+        } catch (InterruptedException e) {
+          return StatusUtils.TIME_OUT;
+        }
       }
-      return status.get() == null ? StatusUtils.TIME_OUT : status.get();
-    } catch (IOException | TException e) {
-      TSStatus status = StatusUtils.INTERNAL_ERROR.deepCopy();
-      status.setMessage(e.getMessage());
-      logger.error("{}: encountered an error when forwarding {} to {}", name, plan, receiver, e);
-      return status;
-    } catch (InterruptedException e) {
-      return StatusUtils.TIME_OUT;
-    }
-  }
 
-  /**
-   * Only the group leader can call this method. Will commit the log locally and send it to
-   * followers
-   *
-   * @param plan
-   * @return OK if over half of the followers accept the log or null if the leadership is lost
-   * during the appending
-   */
-  TSStatus processPlanLocally(PhysicalPlan plan) {
-    logger.debug("{}: Processing plan {}", name, plan);
-    if (readOnly) {
-      return StatusUtils.NODE_READ_ONLY;
-    }
+      /**
+       <<<<<<< HEAD
+       * Only the group leader can call this method. Will commit the log locally and send it to
+       * followers
+       *
+       =======
+       * Create a log for "plan" and append it locally and to all followers.
+       * Only the group leader can call this method.
+       * Will commit the log locally and send it to followers
+       >>>>>>> 378da6117d05a7e38d05fb46cb20b3023434c08c
```

Comment:
Conflict marks.

Reply:
Thanks for your reminding. I will remove them.